File: Review-07-Java-Toys/Example-3/src/main/java/edu/odu/cs/cs330/examples/EnrollStudents.java
package edu.odu.cs.cs330.examples;

import edu.odu.cs.cs350.examples.Student;
import edu.odu.cs.cs350.examples.Roster;

class EnrollStudents {
    /**
     * This is a non-trivial main function 
     */
    public static void main(String args[])
    {
        Student john  = new Student("John");
        Student tom   = new Student("Tom");
        Student jay   = new Student("Jay");
        Student oscar = new Student("Oscar");

        Student[] allStudents = {john, tom, jay, oscar};

        Roster cs330 = new Roster(3, "CS 330");

        for (Student s : allStudents) {

            if(cs330.enroll(s)) {
                System.out.println(s + " enrolled in " + cs330.getCourseNum());
            }
            else {
                System.out.println(s + " NOT enrolled in " + cs330.getCourseNum());
            }
        }

        System.out.println();

        System.out.println(cs330);
    }
}

File: Review-07-Comparing-Languages/Example-3-Python/enroll_students.py
from examples.roster import Roster
from examples.student import Student

"""
Demonstrate simple Roster and Student ADTs.
"""


def main():
    """
    This is a non-trivial main function.
    """

    john = Student("John")
    tom = Student("Tom")
    jay = Student("Jay")
    oscar = Student("Oscar")

    all_students = [john, tom, jay, oscar]

    cs330 = Roster(3, "CS 330")

    for s in all_students:
        if cs330.enroll(s):
            print(f"{s} enrolled in {cs330.course_num}")

        else:
            print(f"{s} NOT enrolled in {cs330.course_num}")

    print()
    print(cs330)


if __name__ == "__main__":
    main()
